### ISSUE TITLE
Rule Change: [multiline-comment-style] Allow //- slash hyphen blocks

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [eslint/eslint#18122]
### What rule do you want to change?

multiline-comment-style

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
//--------
// Example
//--------
```

### What does the rule currently do for this code?

Reports a complaint:

```plaintext
Error 4:1 Expected a block comment instead of consecutive line comments
```

### What will the rule do after it's changed?

With a new option set, it would allow this style.

...what is the term for this style? I couldn't find anything from searching around...

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

This request is split out of https://github.com/eslint/eslint/pull/18005#discussion_r1458066983. ESLint itself uses this style frequently, and has a [`tools/internal-rules/multiline-comment-style.js`](https://github.com/eslint/eslint/blob/9a4ae3b68a1afd9483d331997635727fb19a1a99/tools/internal-rules/multiline-comment-style.js) to allow it!

I personally avoid these kinds of block delineations. So my preference would be to remove them altogether in favor of leaner source files 😄. But if the ESLint repository has a preference for keeping them, then I think that'd be enough evidence that the core rule should support them.

**Comment by Tanujkanti4441** [comment]
Hi @JoshuaKGoldberg, i noticed `seperate-lines` option is not reporting these comments. [playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG11bHRpbGluZS1jb21tZW50LXN0eWxlOiBbXCJlcnJvclwiLCBcInNlcGFyYXRlLWxpbmVzXCJdICovXG5cbi8vLS0tLS0tLS1cbi8vIEV4YW1wbGVcbi8vLS0tLS0tLS1cbmZvbygpO1xuIiwib3B0aW9ucyI6eyJlbnYiOnt9LCJydWxlcyI6e30sInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fSwiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoibW9kdWxlIn19fQ==)

is this resolves the issue?

**Comment by JoshuaKGoldberg** [comment]
I'm not sure that that's enough. `separate-lines` is a bit more lenient. It'll allow any consecutive line comments - which I'm guessing isn't intended to be allowed for ESLint core? [playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG11bHRpbGluZS1jb21tZW50LXN0eWxlOiBbXCJlcnJvclwiLCBcInNlcGFyYXRlLWxpbmVzXCJdICovXG5cbi8vLS0tLS0tLS1cbi8vIEV4YW1wbGVcbi8vLS0tLS0tLS1cbmZvbygpO1xuXG4vLyBUaGlzIGxpbmVcbi8vIGNhbGxzIGZvbygpXG5mb28oKTtcbiIsIm9wdGlvbnMiOnsiZW52Ijp7fSwicnVsZXMiOnt9LCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e30sImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6Im1vZHVsZSJ9fX0= "playground showing separate-lines)

```js
/* eslint multiline-comment-style: ["error", "separate-lines"] */

//--------
// Example
//--------
foo();

// This line
// calls foo()
foo();
```

**Comment by Tanujkanti4441** [comment]
@JoshuaKGoldberg, But don't you think that instead of being an inbuilt feature these comments are just stylistic preferences.

**Comment by JoshuaKGoldberg** [comment]
Well, yes, I personally think this is a stylistic preference - and I personally don't hold this preference 😄.

But the fact that the ESLint codebase holds the preference, and has a forked/wrapped version of the `multiline-comment-style` rule, in my opinion implies one of the following must be true:

* This is a reasonable enough stylistic preference that it should be built into the core rule
* This is an older and/or usual stylistic preference that doesn't warrant a core rule option (and thus the ESLint codebase should probably migrate off of it)

(strong caveat that those are my assumptions & opinions, and I'm seeking validation of them)

**Comment by dctalbot** [comment]
Seems like this would violate the stated policy WRT stylistic rules https://eslint.org/blog/2020/05/changes-to-rules-policies/
> Generate fewer warnings

Instead of another option, maybe better to make the rule more lenient by default? Fewer options means less maintenance

**Comment by nzakas** [comment]
This is a stylistic rule, and our policy is that we don't update stylistic rules with new options or defaults.
